File: spring/src/main/java/com/example/crowdControl/models/Visitor.java
package com.example.crowdControl.models;

import lombok.Data;
import javax.persistence.*;
import java.util.List;

@Entity
@Data
public class Visitor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int visitorId;

    @OneToMany(mappedBy = "visitor", fetch = FetchType.LAZY)
    private List<Visit> visits;

    private String name;
    private String nric;


}

File: angular/src/app/models/Visitor.ts
import { IVisit } from './Visit';

export interface IVisitor {
    visitorId: number,
    visits: IVisit[],
    name?: string,
    nric: string
}

export class Visitor implements IVisitor {
    visitorId: number;
    visits: IVisit[];
    name: string;
    nric: string
}